test/src/test/java/hudson/model/ItemGroupMixInTest.java

```diff
@@ -207,4 +207,28 @@ public boolean isApplicable(Class jobType) {
         assertThat(Items.getConfigFile(foo).asString(), containsString("<description/>"));
     }
 
+    @Test
+    public void createProjectFromXMLWithSpaces() throws IOException {
+        final String xml = "<?xml version='1.0' encoding='UTF-8'?>\n" +
+                "<!DOCTYPE project[\n" +
+                "  <!ENTITY foo SYSTEM \"file:///\">\n" +
+                "]>\n" +
+                "<project>\n" +
+                "  <actions/>\n" +
+                "  <description>&foo;</description>\n" +
+                "  <keepDependencies>false</keepDependencies>\n" +
+                "  <properties/>\n" +
+                "  <scm class=\"hudson.scm.NullSCM\"/>\n" +
+                "  <canRoam>true</canRoam>\n" +
+                "  <triggers/>\n" +
+                "  <builders/>\n" +
+                "  <publishers/>\n" +
+                "  <buildWrappers/>\n" +
+                "</project>";
+
+        Item foo = r.jenkins.createProjectFromXML("foo bar", new ByteArrayInputStream(xml.getBytes()));
+        // if no exception then JAXP is swallowing these - so there should be no entity in the description.
+        assertThat(Items.getConfigFile(foo).asString(), containsString("<description/>"));
```

Comment:
test the description is correct after replacing the entity with a textual description